### PR TITLE
chore: initial fix for ruff ANN401

### PR DESCRIPTION
**Pull request by chemelli74**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new exception class `DataSignWrongType` to handle incorrect data types during signing.

- **Improvements**
  - Enhanced type safety for various methods and functions, improving code reliability and maintainability.
  - Updated method signatures and return types to provide clearer and more specific type expectations.

These improvements ensure better error handling and more predictable behavior within the application.

### CODE REVIEW

File: midealocal/security.py [hunk 1]
```diff
@@ -212,7 +212,9 @@ class MideaAirSecurity(CloudSecurity):
     def __init__(self, login_key: str) -> None:
         super().__init__(login_key, None, None)
 
-    def sign(self, url: str, data: Any, random: str) -> str:
+    def sign(self, url: str, data: dict[str, Any] | str, random: str) -> str:
+        if isinstance(data, str):
```
Comment (rokam) [inline midealocal/security.py:216]: This should be outside of this PR

File: midealocal/devices/__init__.py [hunk 2]
```diff
@@ -126,7 +127,7 @@ def device_selector(
     model: str,
     subtype: int,
     customize: str,
-) -> Any:
+) -> MideaDevice:
```
Comment (rokam) [inline midealocal/devices/__init__.py:130]: Maybe we should deprecate this method. I don't like this dynamic importing. What is you thought about this?

**Review by rokam — CHANGES_REQUESTED** [review]

File: midealocal/devices/__init__.py (hunk 2 above)
Comment (chemelli74) [reply midealocal/devices/__init__.py:130]: I agree, would you please take care of it ?

File: midealocal/security.py (hunk 1 above)
Comment (chemelli74) [reply midealocal/security.py:222]: SO you prefer a single PR with the check and after this typing improvement ?

File: midealocal/security.py [hunk 3]
```diff
@@ -218,7 +218,9 @@ class MideaAirSecurity(CloudSecurity):
     def __init__(self, login_key: str) -> None:
         super().__init__(login_key, None, None)
 
-    def sign(self, url: str, data: Any, random: str) -> str:
+    def sign(self, url: str, data: dict[str, Any] | str, random: str) -> str:
+        if isinstance(data, str):
+            raise Exception("wrong data type to sign")
```
Comment (rokam) [inline midealocal/security.py:223]: Need to change to a specific exception.

**Review by rokam — CHANGES_REQUESTED** [review]

File: midealocal/device.py [hunk 4]
```diff
@@ -542,8 +542,7 @@ def set_attribute(self, attr: str, value: Any) -> None:
         """Set attribute."""
         raise NotImplementedError
 
-    def get_attribute(self, attr: str) -> Any:
-        """Get attribute."""
+    def get_attribute(self, attr: str) -> bool | int | str | None:
```
Comment (rokam) [inline midealocal/device.py:545]:
```suggestion
    def get_attribute(self, attr: str) -> bool | int | str | None:
    """Get attribute."""
```

File: midealocal/device.py [hunk 5]
```diff
@@ -542,8 +542,8 @@ def set_attribute(self, attr: str, value: Any) -> None:
         """Set attribute."""
         raise NotImplementedError
 
-    def get_attribute(self, attr: str) -> Any:
-        """Get attribute."""
+    def get_attribute(self, attr: str) -> bool | int | str | None:
+    """Get attribute."""
```
Comment (rokam) [inline midealocal/device.py:546]:
```suggestion
        """Get attribute."""

```

**Review by rokam — APPROVED** [review]

**Pull request merged**